### PR TITLE
Adds discovered/found repos to local path map

### LINKED ISSUE #3555: Update shared repo mapping file when you open a repo
When GitLens discovers an open repo, it should automatically add it as an entry to the shared repo disk mapping file (using all of its remotes, its first commit sha, etc.) if it’s not in there already.

Currently it sometimes prompts you to locate a repo even though you already opened it in the past, which feels weird.

### PR DESCRIPTION
**Pull request by axosoft-ramint**
Closes #3555

### CODE REVIEW

File: src/git/gitProviderService.ts [hunk 1]
```diff
@@ -655,6 +655,16 @@ export class GitProviderService implements Disposable {
...
+				for (const addedRepo of added) {
+					try {
+						await this.container.repositoryIdentity.addFoundRepositoryToMap(
+							addedRepo,
+							await this.container.repositoryIdentity.getRepositoryIdentity(addedRepo),
+						);
+					} catch (ex) {
+						Logger.error('Could not add discovered repository to identity map: ', ex);
+					}
+				}
```
Comment (axosoft-ramint) [inline src/git/gitProviderService.ts:667]: @eamodio What are your thoughts on me putting this logic in its own function and just using `queueMicrotask` for it here and in the other place below?

Comment (axosoft-ramint) [reply src/git/gitProviderService.ts]: Ended up doing it, and added gating: https://github.com/gitkraken/vscode-gitlens/pull/3633/commits/7130c687ca4c996a609dfa59c39d7772c392ad49

File: src/git/gitProviderService.ts [hunk 2]
```diff
@@ -2440,6 +2442,22 @@ export class GitProviderService implements Disposable {
...
+	@gate()
+	@log()
+	async writeAddedReposToPathMap(repos: Repository[]): Promise<void> {
+		const scope = getLogScope();
+		for (const repo of repos) {
+			try {
+				await this.container.repositoryIdentity.addFoundRepositoryToMap(
+					repo,
+					await this.container.repositoryIdentity.getRepositoryIdentity(repo),
+				);
```
Comment (eamodio) [inline src/git/gitProviderService.ts:2454]: Can we hide the need to call `getRepositoryIdentity` into the `addFoundRepositoryToMap` call, or at least allow it, but have that method do it if missing.

File: src/plus/repos/repositoryIdentityService.ts [hunk 3]
```diff
@@ -25,6 +25,16 @@ export class RepositoryIdentityService implements Disposable {
 		return this.locateRepository(identity, options);
 	}
 
+	async getRepositoryIdentity(repository: Repository): Promise<RepositoryIdentityDescriptor> {
+		const bestRemote = await this.container.git.getBestRemoteWithProvider(repository.uri);
+		return {
+			name: repository.name,
+			initialCommitSha: await this.container.git.getFirstCommitSha(repository.uri),
```
Comment (eamodio) [inline src/plus/repos/repositoryIdentityService.ts:32]: Please execute these 2 in parallel

**Review by eamodio — CHANGES_REQUESTED** [review]
Looks good -- minor tweaks requested

**Comment by axosoft-ramint** [comment]
@eamodio Improvements you mentioned are in. Ready to review again when you're able. Thanks!

**Review by eamodio — APPROVED** [review]
Added some tweaks (which can be squashed). So, if it looks good to you, it's ready to go.

**Pull request merged**